polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -102,6 +147,14 @@ bool isCandidateLoopNest(LoopLikeOpInterface loop) {
   return true;
 }
 
+/// Get the argument of \p func with nd_item type.
+Value getNdItemArgument(FunctionOpInterface func) {
+  Value lastArg = func.getArguments().back();
```

Comment:
Is possible func will have no args, assert that func has args (or return Value()) when it does not.

Reply:
We already checked that in `isCandidateFunction`. I can still add an assert.